### DISCUSSION TITLE
World Clock

### DISCUSSION THREAD

**Discussion by calhix** [InfiniTimeOrg/InfiniTime#1610] [Ideas]
### Verification

- [X] I searched for similar feature request and found none was relevant.

### Pitch us your idea!

An app to to be able to select multiple time zones to track

### Description

I think an app for a world clock would be really useful. I have family that live in different countries that I contact on a regular basis and being able to see their times from the watch would be great. Perhaps you could have a list of major cities in each timezone or the UTC times you could pick from to track.

**Comment by callingmybluff** [comment]
I would like to work on this issue, but it is currently blocked by #1189, however, I am considering a not-so-neat solution.

**Comment by callingmybluff** [comment]
With the restrictions around using the timezone, I make this little approach, please let me know how you find it.

- [x] Screen to specify the time difference.
- [x] Scroll between the different time zones that were added
- [x] Option of deleting time zones
The concept is ready, I just need to do
- [ ] Persist the list of time zones
- [ ] Display a watch instead of the number only
Attached is a video example

https://user-images.githubusercontent.com/38957246/176782874-7985b238-080a-4bb0-9943-e050c775518b.mp4

**Comment by medeyko** [comment]
I missed the multiple timezones display a lot, but I guess that there are too few memory left on the device to include this feature...

**Comment by callingmybluff** [comment]
Can we then please close tickets that can not be added?

**Comment by FintasticMan** [comment]
They will be able to be added at some point, when it's possible to load apps from the external flash. There is some really good progress on that from JF, who's got images and fonts to load at a pretty good speed.

**Comment by calhix** [comment]
Good work so far, something I was thinking of is akin to the style of the world clock on the apple watch. It would be nice for the first screen in the app to be the list of cities you have selected to track (similar to the fourth apple watch image from the link below). You would then maybe swipe left to a second screen which would have a list of major cities within each time zone (perhaps this would be a good start but it's missing quite a few cities imo: https://i.redd.it/imqp9i619a421.png) with a box to check beside each (similar to the Wake Up app). Or instead of a second screen to swipe to, you could have the list of major cities to track in the quick settings.

You could then select one of the cities in the main app to then bring up a separate screen similar to the first three images of the apple watch link below, having the world map image isn't a big deal but having the sunset/sundown times/other info would be nice.

Perhaps optimally we would have a world clock integrated with gadgetbridge, where you could select the cities you want to track from there to then display within the world clock app on the pinetime.

Here is the icon we can use for it: https://fontawesome.com/icons/globe?s=solid
Apple watch: https://9to5mac.com/wp-content/uploads/sites/6/2018/10/how-to-use-apple-watch-iphone-world-clock-31.jpg?quality=82&strip=all
